[ADD] l10n_ar: Fix identification type of Anonymous Final Consumer partner

zaoral opened the pull request:
### Description of the issue/feature this PR addresses:

The purpose of this partner is that we do not need to report to AFIP the identification number.

### Current behavior before PR:

Since this partner identification was configured as DNI it was throwing an error when validate an invoice for that partner in AFIP, AFIP was requesting us to sent the DNI number.

### Desired behavior after PR is merged:

Now we activate the Sigd "Sin identificar/venta global diaria" identification type (code 99), and proper configure the Anonymous Final Consumer partner:  AFIP validate the invoice directly and does not request us the identification number.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

zaoral commented:
Hi @jco-odoo  @ren-odoo @jjscarafia 

I have discover this bad configuration of the Anonymous Final Consumer partner, please read the PR descripcion for more information.

Best Regards

ren-odoo commented:
Hi @jjscarafia , just to confirm, is the  it_Sigd gonna be sent only when document type is empty and AFIP responsibility type is End Consumer, right? (If the responsibility type is different I think we shouldn't send a default value). 

cc. @zaoral 
Thank you.

jjscarafia commented:
Hi @ren-odoo . That seams good to me, to summarize the idea would be:

1. document type is empty and AFIP responsibility type is End Consumer send "it_Sigd" on electronic invoice
2. that way we make it easier for users to encode new customers, they can just create with afip responsability "End customer"
3. we still enable by default "it_Sigd" just in case someone wants to send that type for another type of responsability
4. we sill set "it_Sigd" by default to "Anonymous Final Consumer"

So, if you agree with this, this PR would be ok for me, we just consider "1" on l10n_ar_edi PR

zaoral commented:
Hi @ren-odoo !

Please let me know if you agreed with the last comments of @jjscarafia 
If the answer is yes, please make a mention a Jos in order merge it.

Best Regards

ren-odoo commented:
Hi @zaoral I agree with @jjscarafia comments.  

cc. @jco-odoo let us know if this sounds good for you, so we can move forward with the update, testing and merge. 


Regards.

zaoral commented:
Hi @jco-odoo, remember this PR

Best Regards

jco-odoo commented:
@ren-odoo @zaoral @jjscarafia Is this the right fix?  We also allow normally not to fill in a VAT number e.g.

zaoral commented:
@jco-odoo

For me it is,  this are the cases.

* If a customer has not identification type and it is a final consumer we should send identification type Sigd (which means Without Category / Global sale): using this tye let us to validate the invoice in the AFIP without having the Identification Number.  This is has been done in l10n_ar_edi.
* For invoices to final consumers that have small total amounts it depends of the company internal policies:
    * If the company really do not want to have to save the info of all the partners and if they do nothing with the partner info then they use this generic partner named "Anonymous Final Consumer" with is already configure with identification type Sigd and with this type we do not need to the sent the identification number to AFIP. This PR make this work properly because formerly the Anonymous Final Consumer has the DNI identification type and AFIP was requesting us to sent a DNI number to be able to validate the invoice in the AFIP.
    * If the company want to create one partner for each final consumer, because want to follow some control of the sale to a user, or example want to sent to the user an email with the invoice, sale order,  etc, the options are:
      
       * Use Sigd type: the company does not care about the partner Identification type ID and number. Using this type will let to report in AFIP without specifying the ID number
       * Use DNI type, this could be because the company care to have this value **or because the invoice total amount  if the invoice is greater than the defined limit by AFIP. and AFIP require us to  to sent the both customer ID type and number**
 
* If a customer has not configured an identification type and is not a final consumer we would not sent the ID number and in this case AFIP will return an error telling us that we need to send the identification type and number.

@jjscarafia I am not sure If you are agree with the above cases or if there is anything missing let us known.

Best Regards

jco-odoo commented:
@zaoral An alternative is to use an xml_id for the final customer and use that one and make it pass the SigD in that case.  (although sigd should be active then too)  I even would prefer to keep the partner empty in those cases, but you need one when validating.

zaoral commented:
@jco-odoo 

Yes, we need a partner and we also need a partner identification.

@jco-odoo @ren-odoo @jjscarafia 

If you like we can activate the SigD type but do not add it to the Final Consumer partner, because in the l10n_ar_edi side we are already taking into account that if a partner has final consumer responsibility ad not identification type we sent to AFIP the SigD type. This way we avoid also to use xml ids. Also I am thinking that a default value for the AFIP responsibility could help the user to report new users, but not sure...

zaoral commented:
Hi @jco-odoo 

I have added the last changes we agree in the last meeting.
I added a big sequence number to the SigD identification type to show it in the bottom of identification type list.

Please review and let me know if there is anything missing in order to merge.

Best Regards

jco-odoo commented:
@robodoo r+
